src/adapter/graphql/src/mutations/dataset_metadata_mut.rs

```diff
@@ -71,7 +80,7 @@ impl DatasetMetadataMut {
         let dataset = self.get_dataset(ctx).await?;
 
         let old_attachments = self
-            .get_last_block_of_type::<odf::SetAttachments>(ctx)
+            .get_last_block_of_type(ctx, <SearchSetAttachmentsVisitor>::default())
```

Comment:
Yes, it looks a bit unfamiliar and in my opinion too, but it's the only way to engage the default error type (`InternalError`).

For example:
```rust
pub type SearchSetVocabVisitor<E = InternalError> =
    SearchSingleTypedBlockVisitor<SetVocab, E, { Flag::SET_VOCAB.bits() }>;
```

Reply:
What is the idea behind `<E = InternalError>`? 

When I'm using `SearchSetVocabVisitor` - the visitor is a concrete type - it should *tell me* what errors it can return... I don't understand how this error type can be overridable.

Reply:
> When I'm using SearchSetVocabVisitor - the visitor is a concrete type - it should tell me what errors it can return

This Visitor does not return any errors. Basically, we need this for calls with collections of Visitors (`accept_*()` methods). Thus, we can combine different translatable error types into one another.

--

Btw, After re-reading this comment, I tried removing `E` but could not compile in the end.